File: backend/PulseGym.Logic/DTO/WorkoutDTO/WorkoutViewDTO.cs
using PulseGym.Entities.Enums;

namespace PulseGym.Logic.DTO
{
    public class WorkoutViewDTO
    {
        public Guid Id { get; set; }

        public DateTime WorkoutDateTime { get; set; }

        public string? Notes { get; set; }

        public required WorkoutStatus Status { get; set; }

        public required WorkoutType WorkoutType { get; set; }

        public TrainerViewDTO? Trainer { get; set; }

        public GroupClassViewDTO? GroupClass { get; set; }

        public ICollection<ClientViewDTO>? Clients { get; set; }
    }
}

File: frontend/src/app/models/workout/workout-view-dto.ts
import { WorkoutStatus } from 'src/app/enums/workout-status';
import { WorkoutType } from 'src/app/enums/workout-type';
import { TrainerViewDTO } from '../trainer/trainer-view-dto';
import { ClientViewDTO } from '../client/client-view-dto';
import { GroupClassViewDTO } from '../group-class/group-class-view-dto';

export interface WorkoutViewDTO {
  id: string;
  workoutDateTime: string;
  notes: string | null;
  status: WorkoutStatus;
  workoutType: WorkoutType;
  trainer: TrainerViewDTO | null;
  groupClass: GroupClassViewDTO | null;
  clients: ClientViewDTO[];
}
